sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -1123,7 +1123,7 @@ case class SubstringIndex(strExpr: Expression, delimExpr: Expression, countExpr:
        4
       > SELECT _FUNC_('bar', 'foobarbar', 5);
        7
-      > SELECT POSITION('bar' IN 'foobarbar');
+      > SELECT _FUNC_('bar' IN 'foobarbar');
```

Comment:
The last example changed from `SELECT POSITION('bar' IN 'foobarbar');` to `SELECT locate('bar' IN 'foobarbar');`? It seems we do not support this syntax:
```sql
spark-sql> SELECT locate('bar' IN 'foobarbar');
Error in query:
mismatched input 'IN' expecting {')', ','}(line 1, pos 20)

== SQL ==
SELECT locate('bar' IN 'foobarbar')
--------------------^^^
```

Reply:
Will revert it and add to ignore set. Thanks.

Reply:
I think it makes sense to write a test which executes all examples and compares results. WDYT?

Reply:
+1

Reply:
https://issues.apache.org/jira/browse/SPARK-29242